src/create.rs

```diff
@@ -12,13 +14,22 @@ pub(crate) const DEFAULT_WALLETS_VAULT_PATH: &str = ".fuel/wallets/";
 /// wallets in the vault.
 pub(crate) fn get_next_wallet_index(dir: &str) -> Result<usize, Error> {
     let sorted_wallets = get_wallets_list(dir)?;
-    Ok(sorted_wallets.last().unwrap().0 + 1)
+    if let Some(last) = sorted_wallets.last() {
+        Ok(last.0 + 1)
+    } else {
+        Ok(0)
+    }
 }
 
-pub(crate) async fn create_wallet(path: Option<PathBuf>) -> Result<(), Error> {
-    // Generate wallet from mnenomic phrase.
-    let mnemonic = Wallet::generate_mnemonic_phrase(&mut rand::thread_rng(), 12).unwrap();
-    let wallet = Wallet::new_from_mnemonic_phrase(&mnemonic, None).unwrap();
+pub(crate) fn new_account(phrase: &str, path: Option<String>) -> Result<(Wallet, String)> {
+    let wallet_path = match &path {
+        Some(path) => PathBuf::from(path),
+        None => home::home_dir().unwrap().join(DEFAULT_WALLETS_VAULT_PATH),
+    };
+    let account_index =
+        get_next_wallet_index(&path.unwrap_or_else(|| DEFAULT_WALLETS_VAULT_PATH.to_string()))?;
+    let derive_path = format!("m/44'/60'/{}'/0/0", account_index);
```

Comment:
Not sure about the coin type, I did take it from an SDK test but I guess we have `1179993420` now. If that is correct I can make the change.